frontend/packages/kubevirt-plugin/src/k8s/enhancedK8sMethods/__tests__/enhancedK8sMethods.spec.ts

```diff
@@ -0,0 +1,205 @@
+    expect(history[idx]).toEqual(expectHistoryItem);
+  });
+};
+
+describe('enhancedK8sMethods.js', () => {
+  const testVM = new VMWrapper()
+    .setModel(VirtualMachineModel)
+    .setName('testVM')
+    .setNamespace('default')
+    .setMemory('5', 'Gi')
+    .asResource();
+
+  const otherTestVM = new VMWrapper()
+    .setModel(VirtualMachineModel)
+    .setName('otherVM')
+    .setNamespace('kube')
+    .setMemory('1', 'Mi')
+    .asResource();
+
+  const testPatch = new PatchBuilder('/')
+    .add({
+      additionalData: 'testData',
+    })
+    .build();
+
+  it('records history and shows actualState', async () => {
+    const { methods, methodsUnchecked } = spyEnhancedK8sMethods();
+
+    expect(methods.getHistory()).toHaveLength(0);
+    expect(methods.getActualState()).toHaveLength(0);
+
+    await methods.k8sCreate(VirtualMachineModel, testVM);
+    await methods.k8sPatch(VirtualMachineModel, testVM, [testPatch]);
+    await methods.k8sKill(VirtualMachineModel, testVM);
+    const history = methods.getHistory();
+
+    expectHistory(history, [
+      new HistoryItem(HistoryType.CREATE, testVM),
+      new HistoryItem(HistoryType.PATCH, { ...testVM, ...testPatch }),
+      new HistoryItem(HistoryType.DELETE, testVM),
+    ]);
+
+    expect(methodsUnchecked.history).toHaveLength(3);
+  });
+
+  it('shows actualState', async () => {
+    const { methods, methodsUnchecked } = spyEnhancedK8sMethods();
+
+    await methods.k8sCreate(VirtualMachineModel, testVM);
+    await methods.k8sPatch(VirtualMachineModel, testVM, []);
+    await methods.k8sPatch(VirtualMachineModel, testVM, []);
+    await methodsUnchecked.k8sGet(VirtualMachineModel, testVM);
+    await methods.k8sKill(VirtualMachineModel, testVM);
+    await methods.k8sCreate(VirtualMachineModel, otherTestVM);
+    await methods.k8sPatch(VirtualMachineModel, otherTestVM, []);
+    await methodsUnchecked.k8sGet(VirtualMachineModel, otherTestVM);
+    await methods.k8sCreate(VirtualMachineModel, testVM);
+    await methodsUnchecked.k8sGet(VirtualMachineModel, testVM);
+    await methods.k8sPatch(VirtualMachineModel, testVM, [testPatch]);
+    const actualState = methods.getActualState();
+
+    expect(actualState).toHaveLength(2);
+    expect(actualState[0]).toEqual(otherTestVM);
+    expect(actualState[1]).toEqual({ ...testVM, ...testPatch });
+  });
+
+  it('disables recording history', async () => {
+    const { methods, methodsUnchecked } = spyEnhancedK8sMethods();
+
+    await methods.k8sCreate(VirtualMachineModel, testVM, null, disableHistoryOpts);
+    await methods.k8sPatch(VirtualMachineModel, testVM, [testPatch], disableHistoryOpts);
+    await methods.k8sKill(VirtualMachineModel, testVM, null, null, disableHistoryOpts);
+
+    expect(methods.getHistory()).toHaveLength(0);
+    expect(methodsUnchecked.history).toHaveLength(0);
+
+    await methods.k8sCreate(VirtualMachineModel, testVM);
+    expect(methods.getHistory()).toHaveLength(1);
```

Comment:
can you swap the expects? so first we check that there is history and then we actually test that there isn't

Reply:
the other way is better - otherwise it could happend that some object overwrites the first one we checked and we wouldn't know a difference